### PR TITLE
WT-11333 Update test/format to generate random compaction "free_space_target" values for foreground compaction

### PR DESCRIPTION
**Pull request by Sean04**

### CODE REVIEW

File: test/format/format_config_def.c
Comment (pmacko86) [inline test/format/format_config_def.c:178]: You should explicitly mention the unit in the description (i.e., "MB").

**Review by pmacko86 — APPROVED** [review]
LGTM, after fixing one small issue (and the Evergreen test failures).

File: test/format/format_config.c
Comment (etienneptl) [inline test/format/format_config.c:39]: Not sure if this one is auto-generated but it does not seem to be A->Z

**Review by etienneptl — APPROVED** [review]

**Comment by Sean04** [comment]
Test format is failing due to the config check missing for the new API. I will wait for WT-11332 before merging this.

**Review by etienneptl — APPROVED** [review]
Nice one!

**Comment by Sean04** [comment]
evergreen merge

**Pull request merged**
